test/test_grocy.py

```diff
@@ -216,6 +216,36 @@ def test_consume_product_error(self):
         )
         self.assertRaises(GrocyError, self.grocy.consume_product, 1, 1.3)
 
+    @responses.activate
+    def test_add_product_by_barcode_valid(self):
```

Comment:
Could you convert these to "cassettes-based" tests please? I'm trying to get rid of `responses` based tests.

Reply:
I've done this and they do pass but I am unfamiliar with the cassette based tests so I may well have done them completely wrong!